src/test/java/org/broadinstitute/consent/http/resources/DatasetResourceTest.java

```diff
@@ -139,6 +139,39 @@ public void testCreateDatasetErrors() {
         assertEquals(400, responseNameInUse.getStatus());
     }
 
+    @Test
+    public void testUpdateDataset() {
+        DataSet preexistingDataset = new DataSet();
+        when(datasetService.getDatasetByName("test")).thenReturn(preexistingDataset);
+        when(datasetService.updateDataset(any(), any(), any())).thenReturn(preexistingDataset);
+        when(authUser.getGoogleUser()).thenReturn(googleUser);
+        when(googleUser.getEmail()).thenReturn("email@email.com");
+        when(userService.findUserByEmail(any())).thenReturn(dacUser);
+        when(dacUser.getDacUserId()).thenReturn(1);
+        when(uriInfo.getRequestUriBuilder()).thenReturn(uriBuilder);
+        when(uriBuilder.replacePath(anyString())).thenReturn(uriBuilder);
+        initResource();
+        Response response = resource.updateDataset(authUser, uriInfo, "{\"properties\":[{\"propertyName\":\"Dataset Name\",\"propertyValue\":\"test\"}]}", 1);
```

Comment:
Instead of passing in a string, it would be cleaner to pass in a Gson.toJson string value constructed from an actual instance of DataSetDTO. If we do that, then any kind of refactoring around DataSetDTO would be picked up in this test method without having to find it by a test failure.